frontend/src/routes/project/[project_code]/AddProjectMember.svelte

```diff
@@ -1,63 +1,45 @@
 <script lang="ts">
 	import { BadgeButton } from '$lib/components/Badges';
-	import Modal, { DialogResponse } from '$lib/components/modals/Modal.svelte';
-	import { lexSuperForm } from '$lib/forms';
-	import Form from '$lib/forms/Form.svelte';
+	import { FormModal } from '$lib/components/modals';
 	import Input from '$lib/forms/Input.svelte';
 	import UserRoleSelect from '$lib/forms/UserRoleSelect.svelte';
 	import { ProjectRole } from '$lib/gql/graphql';
 	import t from '$lib/i18n';
 	import { z } from 'zod';
 	import { _addProjectMember } from './+page';
 
-	const formSchema = z.object({
+	export let projectId: string;
+
+	const schema = {
 		email: z.string().email($t('project_page.add_user.email_required')),
 		role: z.enum([ProjectRole.Editor, ProjectRole.Manager]).default(ProjectRole.Editor),
-	});
+	};
+	let formModal: FormModal<typeof schema>;
+	$: form = formModal?.form();
 
-	export let projectId: string;
-	let modal: Modal;
-	let { form, errors, reset, message, enhance } = lexSuperForm(formSchema, () =>
-		modal.submitModal(),
-	);
 	async function openModal(): Promise<void> {
-		if ((await modal.openModal()) === DialogResponse.Cancel) return;
-		const result = await _addProjectMember({ projectId, userEmail: $form.email, role: $form.role });
-		if (result.error) {
-			$message = result.error.message;
-			// again go back tqo the top and await a response from the modal.
-			return await openModal();
-		}
-		modal.close();
+		await formModal.open(async (form) => {
+			const result = await _addProjectMember({ projectId, userEmail: $form.email, role: $form.role });
+			return result.error?.message;
+		});
 	}
 </script>
 
-<BadgeButton icon="i-mdi-account-plus-outline" type="badge-success" on:click={openModal}>
+<BadgeButton icon="i-mdi-account-plus-outline" on:click={openModal}>
 	{$t('project_page.add_user.add_button')}
 </BadgeButton>
 
-<Modal bind:this={modal} on:close={() => reset()} bottom>
-	<Form id="addProjectMember" {enhance}>
-		<p>{$t('project_page.add_user.modal_title')}</p>
-		<Input
-			id="email"
-			type="email"
-			label={$t('admin_dashboard.column_email')}
-			bind:value={$form.email}
-			required
-			error={$errors.email}
-			autofocus
-		/>
-		<UserRoleSelect bind:value={$form.role} error={$errors.role} />
-	</Form>
-	<svelte:fragment slot="actions" let:closing>
-		{#if $message}
-			<label class="label">
-				<span class="label-text-alt text-lg text-error mb-2">{$message}</span>
-			</label>
-		{/if}
-		<button type="submit" form="addProjectMember" class="btn btn-primary" class:loading={closing}>
-			{$t('project_page.add_user.submit_button')}
-		</button>
-	</svelte:fragment>
-</Modal>
+<FormModal bind:this={formModal} {schema} let:errors>
```

Comment:
could we do a let:form here? or would that break type safety?

Reply:
We could, but we can't `bind:` to it: `Cannot bind to a variable declared with the let: directive`

Reply:
ahh ok, we're certainly finding a lot of edge cases in svelte here lol.